test/api/TradeAPI.spec.ts

```diff
@@ -175,4 +214,314 @@ describe('TradeAPI', () => {
+      subjectToToken = '0xBBBB262A92581EC09C2d522b48bCcd9E3C8ACf9C';
+      subjectFromTokenDecimals = 8;
+      subjectToTokenDecimals = 6;
+      subjectRawAmount = '5';
+      subjectFromAddress = '0xCCCC262A92581EC09C2d522b48bCcd9E3C8ACf9C';
+      subjectSetToken = <unknown>{ val: 'settoken' } as SetTokenAPI;
+      subjectGasPrice = 20;
+    });
+
+    async function subject(): Promise<TradeQuote> {
+      return await tradeAPI.fetchTradeQuoteAsync(
+        subjectFromToken,
+        subjectToToken,
+        subjectFromTokenDecimals,
+        subjectToTokenDecimals,
+        subjectRawAmount,
+        subjectFromAddress,
+        subjectSetToken,
+        subjectGasPrice
+      );
+    }
+
+    it('should call the TradeQuoter with correct params', async () => {
+      const expectedQuoteOptions = {
+        fromToken: subjectFromToken,
+        toToken: subjectToToken,
+        fromTokenDecimals: subjectFromTokenDecimals,
+        toTokenDecimals: subjectToTokenDecimals,
+        rawAmount: subjectRawAmount,
+        fromAddress: subjectFromAddress,
+        chainId: (await provider.getNetwork()).chainId,
+        tradeModule: tradeModuleWrapper,
+        provider: provider,
+        setToken: subjectSetToken,
+        gasPrice: subjectGasPrice,
+        slippagePercentage: undefined,
+        isFirmQuote: undefined,
+        feePercentage: undefined,
+        feeRecipient: undefined,
+        excludedSources: undefined,
+      };
+      await subject();
+
+      expect(tradeQuoter.generate).to.have.beenCalledWith(expectedQuoteOptions);
+    });
+
+    describe('when the fromToken address is invalid', () => {
+      beforeEach(async () => {
+        subjectFromToken = '0xInvalidAddress';
+      });
+
+      it('should throw with invalid params', async () => {
+        await expect(subject()).to.be.rejectedWith('Validation error');
+      });
+    });
+
+    describe('when the toToken address is invalid', () => {
+      beforeEach(async () => {
+        subjectToToken = '0xInvalidAddress';
+      });
+
+      it('should throw with invalid params', async () => {
+        await expect(subject()).to.be.rejectedWith('Validation error');
+      });
+    });
+
+    describe('when the fromTokenDecimals is invalid', () => {
+      beforeEach(async () => {
+        subjectFromTokenDecimals = <unknown>'100' as number;
+      });
+
+      it('should throw with invalid params', async () => {
+        await expect(subject()).to.be.rejectedWith('Validation error');
+      });
+    });
+
+    describe('when the toTokenDecimals is invalid', () => {
+      beforeEach(async () => {
+        subjectToTokenDecimals = <unknown>'100' as number;
+      });
+
+      it('should throw with invalid params', async () => {
+        await expect(subject()).to.be.rejectedWith('Validation error');
+      });
+    });
+
+    describe('when the rawAmount quantity is invalid', () => {
+      beforeEach(async () => {
+        subjectRawAmount = <unknown>5 as string;
```

Comment:
What is `<unknowng>` 👀

Reply:
TS complains if you make an incoherent cast (like number to string) ... you have to cast to unknown so it forgets 5 is a number